mlrun/model_monitoring/controller.py

```diff
@@ -312,107 +308,79 @@ def _parse_batch_dict_str(self):
-            self.batch_dict
-        )
-        threshold_date = schedule_time - datetime.timedelta(days=1)
-        threshold_year = threshold_date.year
-        threshold_month = threshold_date.month
-        threshold_day = threshold_date.day
-
-        base_directory = get_monitoring_parquet_path(
-            project=self.project,
-            kind=mm_constants.FileTargetKind.BATCH_CONTROLLER_PARQUET,
-        )
-        target = ParquetTarget(path=base_directory)
-        store, _ = target._get_store_and_path()
-        fs = store.get_filesystem()
+    def _delete_old_parquet(self, endpoints: List[Dict[str, Any]], days: int = 1):
+        """Delete application parquets that are older than 24 hours.
 
-        try:
-            # List all subdirectories in the base directory
-            years_subdirectories = fs.listdir(base_directory)
-
-            for y_subdirectory in years_subdirectories:
-                year = int(y_subdirectory["name"].split("/")[-1].split("=")[1])
-                if year == threshold_year:
-                    month_subdirectories = fs.listdir(y_subdirectory["name"])
-                    for m_subdirectory in month_subdirectories:
-                        month = int(m_subdirectory["name"].split("/")[-1].split("=")[1])
-                        if month == threshold_month:
-                            day_subdirectories = fs.listdir(m_subdirectory["name"])
-                            for d_subdirectory in day_subdirectories:
-                                day = int(
-                                    d_subdirectory["name"].split("/")[-1].split("=")[1]
-                                )
-                                if day == threshold_day - 1:
-                                    fs.rm(path=d_subdirectory["name"], recursive=True)
-                        elif month == threshold_month - 1 and threshold_day == 1:
-                            fs.rm(path=m_subdirectory["name"], recursive=True)
-                elif (
-                    year == threshold_year - 1
-                    and threshold_month == 1
-                    and threshold_day == 1
-                ):
-                    fs.rm(path=y_subdirectory["name"], recursive=True)
-        except FileNotFoundError as exc:
-            logger.warn(
-                f"Batch application process were unsuccessful to remove the old parquets due to {exc}."
+        :param endpoints: List of dictionaries of model endpoints records.
+        """
+        if self.parquet_directory.startswith("v3io:///"):
+
+            target = mlrun.datastore.targets.BaseStoreTarget(
+                path=self.parquet_directory
+            )
+            store, _ = target._get_store_and_path()
+            fs = store.get_filesystem()
+
+            # calculate time threshold (keep only files from the last 24 hours)
+            time_to_keep = float(
+                (datetime.datetime.now() - datetime.timedelta(days=days)).strftime("%s")
```

Comment:
Question: do we intend to keep using `datetime.datetime.now()` here?
It is plausible unless the system really crashes.

Reply:
I don't think it's an issue (assuming that the controller timezone is not changing after the crash). The controller creates the parquets and manage the removal process based on the same timezone.